src/language/providers/completionProvider.ts

```diff
@@ -0,0 +1,187 @@
+import { CompletionItem, CompletionItemKind, languages } from "vscode";
+import Database, { SQLType } from "../../database/schemas";
+import Table from "../../database/table";
+import Document from "../sql/document";
+import { ObjectRef } from "../sql/types";
+import { updateCache } from "./completionItemCache";
+import CompletionItemCache from "./completionItemCache";
+
+const completionItemCache = new CompletionItemCache();
+
+export interface CompletionType {
+  order: string;
+  label: string;
+  type: SQLType;
+  icon: CompletionItemKind;
+}
+
+const completionTypes: { [index: string]: CompletionType } = {
+  tables: {
+    order: `a`,
+    label: `table`,
+    type: `tables`,
+    icon: CompletionItemKind.File,
+  },
+  views: {
+    order: `b`,
+    label: `view`,
+    type: `views`,
+    icon: CompletionItemKind.Interface,
+  },
+  aliases: {
+    order: `c`,
+    label: `alias`,
+    type: `aliases`,
+    icon: CompletionItemKind.Reference,
+  },
+};
+
+function createCompletionItem(
+  name: string,
+  kind: CompletionItemKind,
+  detail?: string,
+  documentation?: string,
+  sortText?: string
+): CompletionItem {
+  const item = new CompletionItem(name, kind);
+  item.detail = detail;
+  item.documentation = documentation;
+  item.sortText = sortText;
+  return item;
+}
+
+function getColumnAtributes(column: TableColumn): string {
+  return `Field: ${column.COLUMN_NAME}\n Type: ${column.DATA_TYPE}\n HAS_DEFAULT: ${column.HAS_DEFAULT}\n IS_IDENTITY: ${column.IS_IDENTITY}\n IS_NULLABLE: ${column.IS_NULLABLE}`;
```

Comment:
Instead of returning a single string with the use of newline (`\n`), let's build an array of lines and then use `.join('\n')` instead.